src/attr/_compat.py

```diff
@@ -10,6 +10,13 @@
 PYPY = platform.python_implementation() == "PyPy"
 
 
+if PYPY or sys.version_info[:2] >= (3, 6):
```

Comment:
for both pypy and python 3.6 the ordering of dicts is a implementation detail,
i believe for consistency and correctness dict should only be taken as "ordered" once the language declares it a standard instead of a implementation detail

additionally this can create a regression where people create a dict but accidentially rely on slightly different order enabled by the creation counter

Reply:
PyPy was always very open about having ordered dicts and CPython has promoted it to official in 3.7 so I’d find it rather unpragmatic to not rely on it.

As for regression: I thought long and hard but can’t come up with a sensible situation? At worst, it might reveal some hidden bug?

Reply:
thanks for addressing the concerns

wrt the regression, i propose adding a hint to the release notes in order  to help people that do hit it
(i suspect it will be really rare and after your mention of hidden bugs i believe those would be the most likely cause)

Reply:
I would love to add a hint if someone comes up with a realistic scenario. 🤔

Most people did `{"a": attr.ib(), "b": attr.ib()}` which is gonna work just the same.  I guess if someone realized it’s the order and did something like:

```
a = attr.ib()
b = attr.ib()
{"b": b, "a": a}
```

by accident?

P.S. another reminder that pytest should s/convert/converter ;)

Reply:
we are working on the pytest conversion, just really slow, after all we have some bits of code that act incorrect and cant be corrected with attrs easily (we do have to keep compatibility for some time even if we are unhappy about some of the things we have to keep it with)

Reply:
@RonnyPfannschmidt just curious using both attr and pytest (and pluggy ;) ), what is this conversion about?

Reply:
@pombredanne pytest is sowly going to switch to using attrs - its goingto be a long process

Reply:
@RonnyPfannschmidt to be clear: you’re using a deprecated API. I’m not talking about the overall work. :)

Reply:
@hynek oh, i see, i will check what attrs version to pin for the correct api then

Reply:
<19.1 most likely.  But you could also just..like..fix it. :) literally just s/convert/converter/.

Reply:
17.4

Reply:
thanks 👍